app/tests/unit/containers/Forms/ProductsArrayField.test.tsx

```diff
@@ -0,0 +1,63 @@
+import React from 'react';
+import {shallow} from 'enzyme';
+import {ProductsArrayField} from 'containers/Forms/ProductsArrayField';
+import {
+  ProductsArrayField_naicsProducts,
+  CiipProductState
+} from 'ProductsArrayField_naicsProducts.graphql';
+import {createDefaultJsonSchemaFormProps} from 'tests/json-schema-utils';
+
+const getTestQuery = () => {
+  return {
+    ' $refType': 'ProductsArrayField_naicsProducts',
+    mandatoryProducts: {
+      edges: [
+        {
+          node: {
+            id: 'abc',
+            productByProductId: {
+              rowId: 9,
+              units: 'GWH',
+              productState: 'PUBLISHED' as CiipProductState,
+              requiresEmissionAllocation: true,
+              requiresProductAmount: true,
+              isEnergyProduct: false,
+              addPurchasedElectricityEmissions: false,
+              addPurchasedHeatEmissions: false,
+              subtractExportedElectricityEmissions: false,
+              subtractExportedHeatEmissions: false,
+              subtractGeneratedElectricityEmissions: false,
+              subtractGeneratedHeatEmissions: false,
+              addEmissionsFromEios: false
+            }
+          }
+        }
+      ]
+    }
+  };
+};
+
+const getTestElement = ({noNaicsCode = false, relay = null}) => {
+  const data = getTestQuery();
+  return (
+    <ProductsArrayField
+      {...createDefaultJsonSchemaFormProps()}
+      naicsProducts={
+        noNaicsCode ? null : (data as ProductsArrayField_naicsProducts)
+      }
+      relay={relay}
+    />
+  );
+};
+
+describe('ProductsArrayField', () => {
+  it('no NAICS code specified', () => {
+    const wrapper = shallow(getTestElement({noNaicsCode: true}));
+    expect(wrapper.find('Relay(ProductField)').exists()).toBeFalse();
+    expect(wrapper).toMatchSnapshot();
+  });
+  it('NAICS code specified, with allowable products', () => {
+    const wrapper = shallow(getTestElement({}));
+    expect(wrapper).toMatchSnapshot();
+  });
```

Comment:
It would be nice to add more assertions... Since the component relies on there being a formResult added to the formContext prop, we'd need to modify the `createDefaultJsonSchemaFormProps` helper in json-schema-utils

Reply:
Using `mount` instead of `shallow` we should be able to test the useEffect function, and spy on the mutation?

We could always pass in a formContext with someting like this
```javascript
const props = {...createDefaultJsonSchemaFormProps(), formContext: {ciipFormResult: {...etc...}}}
```